cluster-autoscaler/cloudprovider/clusterapi/clusterapi_controller.go

```diff
@@ -299,34 +302,38 @@ func newMachineController(
 	informerFactory := dynamicinformer.NewFilteredDynamicSharedInformerFactory(dynamicclient, 0, metav1.NamespaceAll, nil)
 
 	CAPIGroup := getCAPIGroup()
+	if CAPIGroup == "" {
+		return nil, fmt.Errorf("could not get a CAPI group")
+	}
```

Comment:
`getCAPIGROUP` can't return an empty string, so this check is redundant

Reply:
mm It does not hurt though, I guess it safe to assume it since `getCAPIGROUP` is well unit tested. Dropped it.